python/tank/path_cache.py

```diff
@@ -586,29 +600,28 @@ def _do_incremental_sync(self, cursor, log, sg_data):
                     - metadata
                     - path 
         """
-
+        
         if len(sg_data) == 0:
             return []
         
+        self._log_debug(log, "Begin replaying FilesystemLocation entities locally...")
+        
         # find the max event log id in sg_data. We will store this in the sync db later.
         max_event_log_id = max( [x["id"] for x in sg_data] )
         
         created_folder_ids = []
         for d in sg_data:
+            self._log_debug(log, "Looking at event log entry %s" % d)
             if d["event_type"] == "Toolkit_Folders_Create":
                 # this is a creation request! Replay it on our database
                 created_folder_ids.extend( d["meta"]["sg_folder_ids"] )
             else:
                 # should never come here
                 raise Exception("Unsupported event type '%s'" % d)
-                
-        if len(created_folder_ids) == 0:
-            # one or more folder creation events were detected but none of them had actually
-            # resulted in any actual folders being created!
-            return []
```

Comment:
Might be worth adding a comment to that effect in the code so that someone doesn't come along at a later date and optimise it!